src/pages/rekey-reason/rekey-reason.html

```diff
@@ -70,7 +111,7 @@ <h4>Reason for use</h4>
 </ion-content>
 <ion-footer>
   <ion-row class="mes-full-width-card box-shadow">
-    <button type="submit" class="mes-primary-button" ion-button (click)="onUploadPressed()">
+    <button [disabled]="!this.formIsValid(false)" type="submit" class="mes-primary-button" ion-button (click)="onUploadPressed()">
```

Comment:
Don't disable the button. Allow the button to be pressed regardless of the page state, then provide feedback on the form if the validation fails.

Reply:
Ammar mentioned that I should disable the button if the form is invalid, I think it was based on PO comments

Reply:
It might be tempting, but it's the wrong approach. It's not a pattern we're using anywhere else in the app and so it would also be introducing inconsistency. The pattern should be, press button, validate, show errors on the page.

With the button disabled, the user is left wondering, how do I get this button to work?

Nice article here for further reading - https://axesslab.com/disabled-buttons-suck/